### PR TITLE
ELUB tolerates extreme LRs

### PR DESCRIPTION
**Pull request by wowtor**
ELUB would throw an error if LRs of 0 or inf occur; this is fixed by substituting them by large values.

### CODE REVIEW

File: lir/bayeserror.py [hunk 1]
```diff
@@ -48,23 +48,35 @@ def elub(lrs, y, add_misleading=1, step_size=.01):
...
-    lower_bound = np.max(eu_negative_left * np.exp(step_size), initial=np.min(lrs))
-    upper_bound = np.min(eu_negative_right / np.exp(step_size), initial=np.max(lrs))
+    lower_bound = np.max(eu_negative_left * np.exp(step_size), initial=np.min(sanitized_lrs))
+    upper_bound = np.min(eu_negative_right / np.exp(step_size), initial=np.max(sanitized_lrs))
 
-    return lower_bound, upper_bound
+    if lower_bound < upper_bound:
+        return lower_bound, upper_bound
+    else:
+        return 1, 1
```
Comment (rolfypma) [inline lir/bayeserror.py]: is dit wat je wilt? niet een exception oid (voor mij niet duidelijk wanneer dit gebeurt)

File: lir/bayeserror.py [hunk 2]
```diff
@@ -48,23 +48,35 @@ def elub(lrs, y, add_misleading=1, step_size=.01):
...
     :param step_size: required accuracy on a natural logarithmic scale
+    :param substitute_for_extremes (tuple of scalars): substitute for extreme LRs, i.e.
+        LRs of 0 and inf are substituted by these values
     """
-    llrs = np.log(lrs)
-    log_lr_threshold_range = (np.min(llrs), np.max(llrs))
+
+    # remove LRs of 0 and infinity
+    sanitized_lrs = lrs
+    sanitized_lrs[sanitized_lrs == 0] = substitute_extremes[0]
```
Comment (rolfypma) [inline lir/bayeserror.py:57]: is dit de enige plek waarop dit nodig is? lijkt me prima om dit te mergen maar als we het elders ook gaan doen is een generiek stukje code misschien aardig

**Review by rolfypma — APPROVED** [review]

File: lir/bayeserror.py (hunk 1 above)
Comment (wowtor) [reply lir/bayeserror.py]: Dit hoort alleen te gebeuren als de expected utility altijd negatief is. Net nog wat extra uitzoekwerk gedaan en tests geschreven voor edge cases.

File: lir/bayeserror.py (hunk 2 above)
Comment (wowtor) [reply lir/bayeserror.py:57]: Mja, ik vind het lelijk om te doen en wil liever doorrekenen met inf en 0. Alleen hier is het echt nodig omdat hij anders stuk gaat.
